actionview/lib/action_view/template.rb

```diff
@@ -8,6 +8,8 @@ module ActionView
   class Template
     extend ActiveSupport::Autoload
 
+    EXPLICIT_LOCALS_REGEX = /\# locals: \((.*)\)/
```

Comment:
I hate myself for only noticing it now, be we may need to rework that regexp a bit:

  - We should ensure it's the only thing on the line, so add `^` and `$`
  - We can probably allow extra whitespaces in a bunch of places, e.g `\s+` instead of just ` `.

Reply:
I tested `/^\s*\#\s+locals:\s+\((.*)\)$/` on https://rubular.com/, and it seem to do the job

Reply:
🤔 the existing magic comment regex does not have the exact line match: https://github.com/rails/rails/blob/main/actionview/lib/action_view.rb#L33. I'm guessing this is to allow it to work with non-ERB templating languages? For example, in haml they look like:

```haml
-# locals: (foo:)
```

(https://makandracards.com/makandra/689-know-your-haml-comments#section-ruby-comments)

I swapped in the `\s+` 👍🏻

Reply:
(and for that matter, in ERB we wouldn't be matching the entire line either:

```erb
<%# locals: (foo:) %>
```

Of course, I'm super rusty on regexes so forgive me if I'm missing something here ❤️

Reply:
Ah right, we're paring the source template, not the generated code, that makes sense. My bad!

Reply:
All good! I _greatly_ appreciate the rigor of your reviews!